cmd/traffic/cmd/manager/mutator/agent_injector.go

```diff
@@ -230,6 +231,27 @@ func needInitContainer(config *agentconfig.Sidecar) bool {
 	return false
 }
 
+func deleteAppContainer(ctx context.Context, pod *core.Pod, config *agentconfig.Sidecar, patches patchOps) patchOps {
```

Comment:
This should delete the containers that are marked for replace, not all containers that can be intercepted.